src/main/scala/viper/gobra/ast/internal/utility/GobraStrategy.scala

```diff
@@ -86,6 +87,24 @@ object GobraStrategy {
       case (e: Mul, Seq(l: Expr, r: Expr)) => Mul(l, r)(meta)
       case (e: Mod, Seq(l: Expr, r: Expr)) => Mod(l, r)(meta)
       case (e: Div, Seq(l: Expr, r: Expr)) => Div(l, r)(meta)
+      case (e: TypeAssertion, Seq(exp: Expr)) => TypeAssertion(exp, e.arg)(meta)
+      case (e: TypeOf, Seq(exp: Expr)) => TypeOf(exp)(meta)
+      case (e: IsComparableInterface, Seq(exp: Expr)) => IsComparableInterface(exp)(meta)
+      case (e: IsComparableType, Seq(exp: Expr)) => IsComparableType(exp)(meta)
+      case (e: ToInterface, Seq(exp: Expr)) => ToInterface(exp, e.typ)(meta)
+      case (e: BoolTExpr, Seq()) => BoolTExpr()(meta)
+      case (e: IntTExpr, Seq()) => IntTExpr(e.kind)(meta)
+      case (e: PermTExpr, Seq()) => PermTExpr()(meta)
+      case (e: DefinedTExpr, Seq()) => DefinedTExpr(e.name)(meta)
+      case (e: PointerTExpr, Seq(elems: Expr)) => PointerTExpr(elems)(meta)
+      case (e: StructTExpr, Seq()) => StructTExpr(e.fields)(meta)
+      case (e: ArrayTExpr, Seq(length: Expr, elems: Expr)) => ArrayTExpr(length, elems)(meta)
+      case (e: SliceTExpr, Seq(elems: Expr)) => SliceTExpr(elems)(meta)
+      case (e: SequenceTExpr, Seq(elems: Expr)) => SequenceTExpr(elems)(meta)
+      case (e: SetTExpr, Seq(elems: Expr)) => SetTExpr(elems)(meta)
+      case (e: MultisetTExpr, Seq(elems: Expr)) => MultisetTExpr(elems)(meta)
+      case (e: OptionTExpr, Seq(elems: Expr)) => OptionTExpr(elems)(meta)
+      case (e: TupleTExpr, Seq(elems: Vector[Expr@unchecked])) => TupleTExpr(elems)(meta)
```

Comment:
The JVM erases type parameters. As a consequence, you cannot match on type parameters. If you do, you have to write '@unchecked'. Essentially, here I am assumding that the sequence will contain a Vector of expressions, but it could also be a Vector of statements.